test/unit/authorizer_test.rb

```diff
@@ -255,4 +255,39 @@ def setup
     assert_includes results, host
     refute results.grep(host).first.readonly?
   end
+
+  test "#find_collection(Host, :permission => :view_hosts, :joined_on: Report) for matching unlimited filter" do
```

Comment:
I'd rather not, as that isn't describing the context of what we're testing, it's just a fixture of sorts.  (This isn't about view_hosts.)